src/structs/Item.js

```diff
@@ -376,6 +384,13 @@ export class Item extends AbstractStruct {
     if (this.parent && this.parent.constructor === ID && this.id.client !== this.parent.client && this.parent.clock >= getState(store, this.parent.client)) {
       return this.parent.client
     }
+    if (this.content.constructor === ContentType && /** @type {ContentType} */ (this.content).type.constructor === YWeakLink) {
```

Comment:
Is this necessary?

If this is a WeakLink, then it could also exist when the type does not exist yet.

Two reasons to remove this condition:
1. This can (potentially) lead to circular dependencies that can't be resolved (`ymap->yweaklink(yarray)` + `yarray->yweaklink(ymap)`).
2. It enables us to possibly extend WeakLinks to types of subdocuments that are eventually loaded (I'm definitely not suggesting to implement this right now)